definitions/npm/react-i18next_v1.10.x/flow_v0.36.x-/react-i18next_v1.10.x.js

```diff
@@ -0,0 +1,35 @@
+import type { TFunction } from 'i18next'
+
+declare module 'react-i18next' {
+  declare type TFunction = TFunction;
+  declare type Locales = string | Array<string>;
+
+  declare type StatelessComponent<P> = (props: P) => ?React$Element<any>;
+
+  declare type Comp<P> = StatelessComponent<P> | Class<React$Component<*, P, *>>;
+
+  declare type Translator<OP, P> = {
+    (component: StatelessComponent<P>): Class<React$Component<void, OP, void>>;
+    <Def, St>(component: Class<React$Component<Def, P, St>>): Class<React$Component<Def, OP, St>>;
```

Comment:
I admittedly don't understand this syntax. It appears to define a function (callable object) that can be called in two different ways, the first taking a stateless component, the other a class component. The `<Def, St>` part is particularly strange. I borrowed it from https://github.com/flowtype/flow-typed/blob/master/definitions/npm/react-redux_v4.x.x/flow_v0.30.x-/react-redux_v4.x.x.js#L35, but couldn't find any other examples in the docs or elsewhere. If anyone could offer an explanation, I would greatly appreciate it.